tests/integration/targets/ec2_snapshot/tasks/main.yml

```diff
@@ -222,11 +220,8 @@
           "tag:Name": '{{ resource_prefix }}'
       register: info_result
 
-    - assert:
-        that:
-          - info_result.snapshots| length == 3
-
-    - ec2_snapshot:
```

Comment:
Why remove this assertion?  Specifically with the addition of check_mode support it's important to know that we only created the expected number of snapshots.